test/e2e/global_catalog_exclusion_e2e_test.go

```diff
@@ -0,0 +1,126 @@
+			},
+		}
+		_ = determinedE2eClient.Create(context.Background(), globalCatalog)
+
+		By("creating a healthy catalog in the test namespace")
+		localCatalogName := genName("good-catsrc-")
+		var err error = nil
+
+		fbcPath := filepath.Join(testdataDir, magicCatalogDir, "fbc_initial.yaml")
+		localCatalog, err = NewMagicCatalogFromFile(determinedE2eClient, testNamespace.GetName(), localCatalogName, fbcPath)
+		Expect(err).To(Succeed())
+
+		// deploy catalog blocks until the catalog has reached a ready state or fails
+		Expect(localCatalog.DeployCatalog(context.Background())).To(Succeed())
+
+		By("checking that the global catalog is broken")
+		// Adding this check here to speed up the test
+		// the global catalog can fail while we wait for the local catalog to get to a ready state
+		e2eTestHelper.AssertCatalogSource(globalCatalog).EventuallyHasGrpcConnectionWithLastConnectionState(connectivity.TransientFailure)
+	})
+
+	AfterEach(func() {
+		TeardownNamespace(testNamespace.GetName())
+	})
+
+	When("a subscription referring to the local catalog is created", func() {
+		var subscription *v1alpha1.Subscription
+
+		BeforeEach(func() {
+			subscription = &v1alpha1.Subscription{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: testNamespace.GetName(),
+					Name:      genName("local-subscription-"),
+				},
+				Spec: &v1alpha1.SubscriptionSpec{
+					CatalogSource:          localCatalog.GetName(),
+					CatalogSourceNamespace: localCatalog.GetNamespace(),
+					Package:                "packageA",
+					Channel:                "stable",
+					InstallPlanApproval:    v1alpha1.ApprovalAutomatic,
+				},
+			}
+
+			By("creating a subscription")
+			_ = determinedE2eClient.Create(context.Background(), subscription)
+		})
+
+		When("the operator group is annotated with olm.operatorframework.io/exclude-global-namespace-resolution=true", func() {
+			It("the broken subscription should resolve and have state UpgradePending", func() {
+				By("checking that the subscription is indeed not resolving")
+				e2eTestHelper.AssertSubscription(subscription).EventuallyContainsConditionWithConditionType(v1alpha1.SubscriptionResolutionFailed)
+
+				By("annotating the operator group with olm.operatorframework.io/exclude-global-namespace-resolution=true")
+				annotatedOperatorGroup := operatorGroup.DeepCopy()
+				_ = determinedE2eClient.Get(context.Background(), k8scontrollerclient.ObjectKeyFromObject(annotatedOperatorGroup), annotatedOperatorGroup)
+
+				if annotatedOperatorGroup.Annotations == nil {
+					annotatedOperatorGroup.Annotations = map[string]string{}
+				}
+
+				annotatedOperatorGroup.Annotations["olm.operatorframework.io/exclude-global-namespace-resolution"] = "true"
+				_ = determinedE2eClient.Update(context.Background(), annotatedOperatorGroup)
+
+				By("checking that the subscription is now resolved")
+				e2eTestHelper.AssertSubscription(subscription).EventuallyHasState(v1alpha1.SubscriptionStateUpgradePending)
```

Comment:
Should we add a check for SubscriptionStateAtLatest as well, to prove definitively that the subscription succeeded? Is `packageA` actually installable or simply a dummy package?